clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -976,6 +974,10 @@ void maybeReconcile() {
             }
 
             revokeAndAssign(resolvedAssignment, assignedTopicIdPartitions, revokedPartitions, addedPartitions);
+        }).whenComplete((__, error) -> {
+            if (error != null) {
+                log.error("Reconciliation failed.", error);
```

Comment:
Is there a reason why we want this log here? We already have the same but inside the `revokeAndAssign`, when reconciliation completes.

Reply:
Nope, the exception handling inside `revokeAndAssign` is only triggered if the revoke and assign future fails. If `revokeAndAssign` fails outside the future (in particular, inside `revokePartitions`, there is logic that may fail), the exception falls through to here and was silently swallowed before, which cost me an hour of debugging to find.

Reply:
Oh I see now, thanks for the explanation.